src/joystick/hidapi/SDL_hidapi_steamdeck.c

```diff
@@ -221,8 +313,8 @@ static SDL_bool HIDAPI_DriverSteamDeck_UpdateDevice(SDL_HIDAPI_Device *device)
     SDL_DriverSteamDeck_Context *ctx = (SDL_DriverSteamDeck_Context *)device->context;
     SDL_Joystick *joystick = NULL;
     int r;
+    SDL_bool result = SDL_FALSE;
```

Comment:
You don't need this variable. You return false below when there's an error, and you should return true from this function if a packet was processed or there was no data available.

Reply:
I removed the variable and return SDL_TRUE now if there was no error.